### DISCUSSION TITLE
UI does not update based on the updateOrAppend of my state.todos

### DISCUSSION THREAD

**Discussion by idelfonsog2** [pointfreeco/swift-composable-architecture#1044] [Q&A] [answered]
The Composable Architecture uses GitHub issues for bugs. For more general discussion and help, please use [GitHub Discussions](https://github.com/pointfreeco/swift-composable-architecture/discussions) or [the Swift forum](https://forums.swift.org/c/related-projects/swift-composable-architecture) first.

Hi there!
I have the following list in my state:

```swift
  var todos: IdentifiedArrayOf<Todo> = [/**omitted**/]

// my reducer has the following action:

case let .todos(id: identity, action: .toggled):
        guard var found = state.todos.filter({ $0.id == identity}).first else {
          return .none
        }
        
        found.isChecked.toggle()
        state.todos.updateOrAppend(found)
        return .none
```

1. But my UI is not reflecting the state change
2. the `.debug()` reducer add-on *does prints* the state change `(true <--> false)` of my list
3. When I print the state of my list all Todos state `isChecked = false`

What am I missing?

**Comment by iampatbrown** [comment, marked as answer]
@idelfonsog2 Can you include a small example so we can reproduce? Would need to see how your View is structured.

**Reply by idelfonsog2** [reply]
I can fork/update the project and post the link here.... but I grabbed the `swift-composoble-artchitecture/Examples/Todos` project and  and modified the `AppState` and `appReducer` case to the above/below

```swift
// AppState
  var todos: IdentifiedArrayOf<Todo> = [
    Todo(description: "completed", id: UUID(), isComplete: false)
  ]


// appReducer
case .todo(id: let identity, action: .checkBoxToggled):
        guard var found = state.todos.filter({ $0.id == identity}).first else {
          return .none
        }
        
        found.isComplete.toggle()
        state.todos.updateOrAppend(found)
        return .none
```

no other changes in the repo example project were made

**Reply by idelfonsog2** [reply]
Let me know if you still need the fork on my account @iampatbrown Thank you for looking into this 🙏

**Reply by iampatbrown** [reply]
If that is the only change you made to the `Todos` example then `isComplete` is being toggled twice. Once by `todoReducer` and once by `appReducer`

https://github.com/pointfreeco/swift-composable-architecture/blob/97c3a498de2b0c0aed96fc3836195ebde720e822/Examples/Todos/Todos/Todo.swift#L18-L22

**Comment by mbrandonw** [comment]
(NB: Pat already answered while posting this, but thought I would go ahead and post it for more information)

Hi @idelfonsog2, there is a domain defined just for a single todo row, and it's the one responsible for flipping the boolean:

https://github.com/pointfreeco/swift-composable-architecture/blob/97c3a498de2b0c0aed96fc3836195ebde720e822/Examples/Todos/Todos/Todo.swift#L20-L22

Then the "todos" domain (i.e. the domain that handles an entire list of todos) layers on additional logic on top of the `.checkBoxToggled` action, such as sorting the todos to put completed ones at the bottom.

So, as Pat mentioned, with your code changes you are toggling the boolean twice. Once in the todo list domain and again in the todos domain.

If you can explain more of what you are trying to explore in this demo app we might be able to answer some questions.

**Reply by idelfonsog2** [reply]
Thank you very much! @mbrandonw I was inferring that behavior from the Reducer declaration after the `todoReducer.forEach` I just wasn't sure if it was the case. Hope this discussion also helps anyone out there!
